flink-table-store-flink/flink-table-store-flink-common/src/main/java/org/apache/flink/table/store/connector/FlinkConnectorOptions.java

```diff
@@ -82,6 +82,20 @@ public class FlinkConnectorOptions {
                                     + "By default, if this option is not defined, the planner will derive the parallelism "
                                     + "for each statement individually by also considering the global configuration.");
 
+    public static final ConfigOption<Boolean> INFER_SCAN_PARALLELISM =
+            ConfigOptions.key("scan.infer-parallelism")
```

Comment:
IMO, user should not configure `scan.infer-parallelism` because inferring parallelism is default behavior that infers parallelism when `scan.parallelism` isn't set, and use `scan.parallelism` when `scan.parallelism` is set.

Reply:
Yes, your suggestion is also a reasonable idea. I guess that add the `scan.parallelism` option is a simple way to let users set the scan  parallelism , but it is not intelligent enough, so we added the parallelism inference.

So another solution is that we can also use infer-parallelism instead of `scan.parallelism`. If we enable parallelism inference, we use split or bucket number to as job parallelism. If not, we use the default parallelism of the flink job.

But now that the system has added the `scan parallelism` option, I am worried that some users who are using this option will be affected, so I do not delete this option  and  add the switch for inferring parallelism . and make its priority higher than `scan.parallelism`.

Reply:
@zhangjun0x01, as you mentioned, could we update the default value description of `scan.parallelism`? IMO, the default value of `scan.parallelism` could use the inferred parallelism, and also custom to config via user.

Reply:
ok，I update it

Reply:
@zhangjun0x01, you should also remove the config option `scan.infer-parallelism`, not only update the description of `scan.parallelism`.